src/metabase/api/action.clj

```diff
@@ -139,4 +143,28 @@
         (db/update! existing-model id type-row))))
   (first (action/actions-with-implicit-params nil :id id)))
 
+(api/defendpoint POST "/:id/public_link"
+  "Generate publicly-accessible links for this Action. Returns UUID to be used in public links. (If this
+  Action has already been shared, it will return the existing public link rather than creating a new one.) Public
+  sharing must be enabled."
+  [id]
+  (api/check-superuser)
```

Comment:
to confirm: we still working on the exact permission model for public actions, requiring admin now just because it's the 1st milestone right?